### ISSUE TITLE
SMB: DiskShare has already been closed; SFTP does not reconnect upon failure

### ISSUE THREAD

**Issue by RayetSena** [frankframework/frankframework#6003]
I created an adapter which is DecosScanstraat it uses `Samba2Listener`. When the adapter started to work it works first time and after a  while it gives a DiskShare has already been closed error.(2nd image)

Also I created different adapters one of them lists the files in the location using `SftpFileSystemSender` (In 2nd image you can see the SFTP error for now I am using `IbisActionJob` to start/stop adapter as a workaround) and then it makes a call to other adapter which is reads the file using `SftpFileSystemSender` and then writes to file using Samba2Pipe. This one is working when the adapter start, after -+10 min it gives a DiskShare has already been closed error(1st image)

**Screenshots**

![MicrosoftTeams-image (1)](https://github.com/frankframework/frankframework/assets/55273803/6856a5c3-1a1e-47f7-afbe-f8b9c8a2a985)

![Capture](https://github.com/frankframework/frankframework/assets/55273803/52e5ee57-d5f3-43d7-9987-d4099fcbe314)

**Configuration**
```
  <Receiver name="DecosScanstraat" pollInterval="150">
            <Samba2Listener
                domainName="${samba_domain}"
                hostname="${scanstraat_samba_hostname}"
                share="${scanstraat_samba_share}"
                authType="NTLM"
                port="${samba_port}"
                username="${samba_username}"
                password="${samba_password}"
                messageType="contents"
                inputFolder="${scanstraat_samba_inputFolder}"
                wildcard="*.pdf"
                minStableTime="10000"
            />
        </Receiver>

<Samba2Pipe name="WriteFiles"
                action="write"
                overwrite="true"
                authType="NTLM"
                username="${samba_username}"
                password="${samba_password}"
                domainName="${samba_domain}"
                hostname="${hubdassen_samba_hostname}"
                port="${samba_port}"
                share="${hubdassen_samba_share}"
            >
                <Param name="filename" sessionKey="inputParam"
                    xpathExpression="concat($inputFolder, '/', $file)">
                    <Param name="file" sessionKey="originalMessage" />
                    <Param name="inputFolder" value="${hubdassen_samba_inputFolder}" />
                </Param>
                <Param name="contents" />
                <Forward name="success" path="Exit" />
            </Samba2Pipe>
```

**Environment**
```
FF version=7.9-20230907.223421
```

**Reporter**
Sena

**Comment by jkosternl** [comment]
An update what exactly we did on your issue @RayetSena 

1. Made the Samba2FileSystem stronger, to close the client as well, when the disk share fails. Already in branches 7.9, 8.0 and master. Should be available from `7.9.0-hotfix8` or higher.
2. Made the SftpFileSystem stronger, so it can handle disconnects when the SFTP server connection is gone. It now verifies the connection every 60 seconds. When the connection is not working anymore, it is rebuild automatically. 
3. Made the SftpFileSystem also understand absolute directory paths. Inside some commands (e.g. folderExists, removeDirectory (with content), it assumed paths to be relative. This code is still in a PR.

**Comment by RayetSena** [comment]
```
<SenderPipe name="ReadFiles">
                <SftpFileSystemSender
                    name="ReadFiles"
                    action="read"
                    host="${hubdassen_ftp_hostname}"
                    port="${hubdassen_ftp_port}"
                    authAlias="sftp_parkeren"
                    remoteDirectory="${hubdassen_ftp_remoteDirectory}"
                    strictHostKeyChecking="false"
                >
                    <Param name="filename" sessionKey="originalMessage" />
                </SftpFileSystemSender>
                <Forward name="success" path="InputParam" />
            </SenderPipe>

            <PutInSessionPipe name="InputParam">
                <Param name="inputParam" value="&lt;dummy/&gt;" />
                <Forward name="success" path="WriteFiles" />
            </PutInSessionPipe>

            <Samba2Pipe name="WriteFiles"
                action="write"
                overwrite="true"
                authType="NTLM"
                authAlias="svc_waf"
                domainName="${samba_domain}"
                hostname="${hubdassen_samba_hostname}"
                port="${samba_port}"
                share="${hubdassen_samba_share}"
            >
                <Param name="filename" sessionKey="inputParam"
                    xpathExpression="concat($inputFolder, '/', $file)">
                    <Param name="file" sessionKey="originalMessage" />
                    <Param name="inputFolder" value="${hubdassen_samba_inputFolder}" />
                </Param>
                <Param name="contents" />
                <Forward name="success" path="Exit" />
            </Samba2Pipe>
```

**Comment by nielsm5** [comment]
The framework has a recovery mechanism built in. on the receiver, set the attribute onError=RECOVER.

**Comment by jkosternl** [comment]
Update about the process: the attribute `onError=RECOVER` only seems to work currently in some specific pipes, so does not help with SFTP/FTP/Samba stuff. 

We did some investigation and we concluded that the error handling in the Java code of the Receiver is not good in this situation and likely in more situations. It was catching the Exception (the actual Samba timeout error) to quickly and not dealing with it in the way the framework is designed.. 
We have a clue about the solution. We are going to do some more thinking about it and trying out. If that works, we're going to add it to the master/8.0/7.9 code base. 
For now, in the configuration there is nothing you can do about it now. We will update you when we have some usable progress.

**Comment by jkosternl** [comment]
@RayetSena, Niels has fixed some exception handling in the Receiver, which was merged in #6310 
We are not totally sure if that fixes the Samba disconnect/reconnect automatically, without the scheduled restarts of the adapter, with a IbisActionJob. Can you use the latest version to see if that helps? 
Secondly, can you also test the SFTP disconnect issues, to see if it reconnects automatically as well?

I just saw that our release mechanism is a bit broken on the release counter.. so if you download a new `7.9.0-hotfix10` from Nexus, you get a new build, which includes the changes. Sorry for that. 
It also has a small improvement to not log many lines inside the SMB library, which floods your log files (while in debug logging).

**Comment by jkosternl** [comment]
Yesterday, Niels talked with Sena about the issues. He introduced a new mechanism which fixes these connect -> check connection -> reconnect issue. It is inside #6338 and it is a wrapper that opens and closes a sender. That should fix these issues for all senders where this mechanism is needed. 

I am removing the connection validation check from the PRs and master now.
